File: Lab5_13_2/Lab5_13_2/Lab5_13_2/MovingMaterialPointXY.cs
using System;
namespace Lab5_13_2
{
    public class MovingMaterialPointXY: MovingMaterialPoint
    {
        public double X0 { get; set; }
        public double Y0 { get; set; }
        public double A1 { get; set; }
        public double A2 { get; set; }

        public MovingMaterialPointXY(double x, double y, double a1, double a2)
        {
            this.X0 = x;
            this.Y0 = y;
            this.A1 = a1;
            this.A2 = a2;
        }

        public override double findX(double t)
        {
            return this.X0 + this.A1 * Math.Sin(t);
        }

        public override double findY(double t)
        {
            return this.Y0 + this.A2 * Math.Cos(t);
        }
    }
}

File: Lab5_13_2/Moving_Material_Point_X_Y.py
import math
from Moving_Material_Point import Moving_Material_Point


class Moving_Material_Point_X_Y(Moving_Material_Point):

    def __init__(self, x0, y0, a1, a2) -> None:
        self.__x0 = x0
        self.__y0 = y0
        self.__a1 = a1
        self.__a2 = a2

    def get_x0(self):
        return self.__x0

    def set_x0(self, value: float):
        self.__x0 = value

    def get_y0(self):
        return self.__y0

    def set_y0(self, value):
        self.__y0 = value

    def get_a1(self):
        return self.__a1

    def set_a1(self, value):
        self.__a1 = value

    def get_a2(self):
        return self.__a2

    def set_a2(self, value):
        self.__a2 = value

    def find_x(self, t: float):
        return self.__x0 + self.__a1 * math.sin(t)

    def find_y(self, t: float):
        return self.__y0 + self.__a2 * math.cos(t)
